src/Compilers/Core/Portable/Diagnostic/Diagnostic.cs

```diff
@@ -476,6 +476,11 @@ internal Diagnostic WithProgrammaticSuppression(ProgrammaticSuppressionInfo prog
         // compatibility
         internal virtual int Code { get { return 0; } }
 
+        /// <summary>
+        /// True if this diagnostic represents the given error code. This does not force resolution of lazy diagnostics and is safe to call at any time.
+        /// </summary>
+        internal virtual bool IsCode(int code) { return false; }
```

Comment:
>internal virtual bool IsCode(int code) { return false; } [](http://example.com/codeflow?start=8&length=56)

I just started reviewing the PR, therefore, I might not be aware of some aspects of the change. However, at the moment, it looks like making this method virtual look unjustified. I think this method can be implemented once and for all by simply comparing ```code``` with ```this.Code```. Given that, I am not even sure if the method is needed at all, because anyone can perform this comparison explicitly where it is needed.

Reply:
I explained why I added it here: https://github.com/dotnet/roslyn/pull/71140#discussion_r1418203807.

Reply:
> I explained why I added it here ...

I suggested an alternative approach there.